src/kubernetes/client/config/mod.rs

```diff
@@ -26,4 +27,7 @@ pub struct Config {
 
     /// The TLS configuration parameters to use at the HTTP client.
     pub tls_options: TlsOptions,
+
+    /// The Proxy configuration at the HTTP client.
+    pub proxy: ProxyConfig,
```

Comment:
That's a good question and would help reduce the likelihood that one component would mess it up. Not all sinks use or support proxy configuration though. Do you think there is a way we could do that but still reject proxy config on components that don't allow it?

Alternatively we allow it everywhere, even when it would be unused.

Reply:
That is the approach I took with `acknowledgements`, in that it is present for all sinks and sources, but ignored (and undocumented) for those that can't use it. If a minority of components do not have support and you want to provide an error, you could add a `ctx.cannot_proxy()?` function call to provide a common error.

Reply:
I completely agree with doing that!
Shouldn't we do the same with `tls`? (and open an issue for that)

Reply:
👍 it does seem like `tls` and `request` are other good candidates for this treatment. We can follow up on those though.

Reply:
Agreed, I was thinking the same thing when I was making that suggestion. The one (probably small) gotcha with `request` is that it can't be parsed before calling the sink's `build` function as each sink has different defaults to add to the config. Still, we could pass the config data in to the basic context and so remove it from each sink's config.